site/app/models/OfficeHoursQueueModel.php

```diff
@@ -318,4 +318,36 @@ public function weekNumToDate($weeknum, $yearnum): string {
     public function statNiceName($name): string {
         return $this->niceNames[$name] ?? $name;
     }
+
+    public function getQueueSocketMessage() {
+        $row = $this->core->getQueries()->getQueueMessage($this->current_queue_state['queue_code']);
+        if ($row['message'] != null) {
+            $current_time = $this->core->getDateTimeNow();
+            $sent_time = new DateTime($row['message_sent_time']);
+            $elapsed_time = $current_time->diff($sent_time);
+            //see if message was sent today
+            if ($elapsed_time->days == 0) {
+                //if less than 2 hours have passed, show the message
+                if ($elapsed_time->h < 2) {
+                    return $row['message'];
+                }
+                else {
+                    return null;
+                }
+            }
+            else {
+                return null;
+            }
```

Comment:
This is very nit-picky, but I think this chain of if statements could be flattened to one. Then you can just default to return null at the end of the function.

Reply:
you're right, that was an extremely ugly if statement, thanks